### PR TITLE
DS-61

### PR DESCRIPTION
**Pull request by cliffcaseyyet**
Added capability to return 400 and detailed spec error object courtesy of Pan when validation fails. Also apparently the server entry point had zero validation at all up until now.

### CODE REVIEW

File: deps.edn [hunk 1]
```diff
@@ -7,7 +7,7 @@
          :exclusions [org.clojure/clojurescript]}
         com.yetanalytics/project-pan
         {:git/url "https://github.com/yetanalytics/project-pan"
-         :sha "811571d52e376a7eb727efaaf1fd4dc08ffb50fc"
+         :sha "e6fc8632843ceb6bb86cb2e6ce1ce7c3ccca9aaa"
```
Comment (FeLungs) [inline deps.edn:10]: may need to be updated again

File: src/server/com/yetanalytics/datasim/server.clj [hunk 2]
```diff
@@ -153,12 +159,15 @@
               (assoc context
                      :response
                      (try
-                       {:status  200
-                        :headers {"Content-Type"        "application/json; charset=utf-8"
-                                  "Content-Disposition" (format "attachment; filename=\"simulation.json\"")}
```
Comment (FeLungs) [inline src/server/com/yetanalytics/datasim/server.clj:158]: was `"Content-Disposition"` removed intentionally?

File: src/server/com/yetanalytics/datasim/server.clj [hunk 3]
```diff
@@ -15,6 +15,9 @@
             [clj-http.client                        :as client]
             [environ.core                           :refer [env]]
             [com.yetanalytics.datasim.sim           :as sim]
+            [com.yetanalytics.datasim.input         :as sinput]
```
Comment (FeLungs) [inline src/server/com/yetanalytics/datasim/server.clj:18]: where does this get used?

**Review by FeLungs — COMMENTED** [review]
see comments

File: deps.edn (hunk 1 above)
Comment (cliffcaseyyet) [reply deps.edn:10]: for sure, as soon as pan is in

File: src/server/com/yetanalytics/datasim/server.clj (hunk 3 above)
Comment (FeLungs) [reply src/server/com/yetanalytics/datasim/server.clj:18]: brings in spec key

File: src/server/com/yetanalytics/datasim/server.clj (hunk 2 above)
Comment (cliffcaseyyet) [reply src/server/com/yetanalytics/datasim/server.clj:158]: So i think this approach was abandoned in the actual implementation. If you look at the FE code it takes the Blob that comes over XHR and then marshalls it into a downloaded file complete with filename.

File: deps.edn (hunk 1 above)
Comment (FeLungs) [reply deps.edn:10]: :+1:

File: src/server/com/yetanalytics/datasim/server.clj (hunk 2 above)
Comment (FeLungs) [reply src/server/com/yetanalytics/datasim/server.clj:158]: :+1:

File: src/server/com/yetanalytics/datasim/server.clj [hunk 4]
```diff
@@ -153,12 +159,15 @@
...
+                             (run-sim! (-> context
+                                           :request
+                                           :multipart-params))]
+                         {:status  (if (fn? response-body)
+                                     200
+                                     400)
+                          :headers {"Content-Type"        "application/json; charset=utf-8"}
+                          :body    response-body})
                        (catch Exception e
                          {:status 400
```
Comment (milt) [inline src/server/com/yetanalytics/datasim/server.clj:172]: Wouldn't this be a 500?

**Review by milt — COMMENTED** [review]
Looks good, but see note

**Pull request merged**